fdmi/plugins/m0scheduler

```diff
@@ -17,128 +17,277 @@
+        print('Failed to get Hare endpoint from the cluster info.', file=sys.stderr)
+        return None
+    if info.get('process_fid') is None:
+        print('Failed to get client process fid from the cluster info.', file=sys.stderr)
+        return None
+    if info.get('local_endpoint') is None:
+        print('Failed to get local client endpoint from the cluster info.', file=sys.stderr)
+        return None
+    if info.get('fdmi_filter_fid') is None:
+        print('Failed to get fdmi filter id from the cluster config.', file=sys.stderr)
+        return None
+
+    print('  {}'.format(info), file=sys.stderr)
+    return info
+
+
+def listen_on_command(command):
+    global processes
+    p = subprocess.Popen(shlex.split(command), stdin=subprocess.PIPE,
+                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if p is None:
+        print('Failed to run {}'.format(command), file=sys.stderr)
         return -1
-    with open('/etc/motr/confd.xc') as f:
-        for s in f.readlines():
-            # example: ' {0x6c| ((^l|1:81), 2, ^l|1:81, "", ^n|1:3, ^v|1:66, [1: "Bucket-Name"], [1: "192.168.12.2@tcp:12345:4:1"])},'
-            if s.startswith(' {0x6c| ((^l|'):
-                s = s[15:]
-                s = s[:s.index(')')]
-                fdmi_filter_fid = f"0x6c00000000000001:{int(s):#x}"
-                break
-    print ("local:", local_endpoint," ha: ", ha_endpoint, " prof: ", profile_fid, " proc: ", process_fid, " fdmi_filter_fid: ", fdmi_filter_fid)
-    return ret
-
-def run_command(command):
-    global process
-    process = subprocess.Popen(shlex.split(command), stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-    while True:
-        output = process.stdout.readline()
+    processes.append(p)
 
-        if output:
+    print('Listening for FDMI events on:\n  {}'.format(command), file=sys.stderr)
+    while True:
+        output = p.stdout.readline()
+        if output is not None:
             fid = "fid"
             record = output.decode("utf-8")
             if fid in record:
-               json_parser(record)
+                json_parser(record)
 
-        if process.poll() is not None:
-            print("Exiting because m0sched exited");
+        rc = p.poll()
+        if rc is not None:
+            print('Exiting because m0sched exited with:', file=sys.stderr)
+            print('  error code = {}'.format(rc), file=sys.stderr)
+            _, error = p.communicate()
+            if error is not None:
+                error = error.decode('utf-8')
+            if error is not None and len(error) > 0:
+                print('  error text = {}'.format(error), file=sys.stderr)
             break
 
-    rc = process.poll()
-    return rc
+    return p.poll()
+
+
+def main(args):
+    global cluster_info
+
+    einval = 22
+    if args.config_path is None and args.filter_id is None:
+        print('One of two --config-path or --filter-id is required.', file=sys.stderr)
```

Comment:
Describe which config file is used in comments and may be more elaborate usage text.

Reply:
Please describe where to find the filter-id from in a cluster.

Reply:
ok